source/common/grpc/async_client_impl.h

```diff
@@ -49,9 +49,10 @@ class AsyncClientImpl final : public AsyncClient<RequestType, ResponseType> {
 
   AsyncStream<RequestType>* start(const Protobuf::MethodDescriptor& service_method,
                                   AsyncStreamCallbacks<ResponseType>& callbacks) override {
+    const Optional<std::chrono::milliseconds> no_timeout;
```

Comment:
A first thought is `FooBar&` for the case when a reference is not retained past the unwinding of the call stack, `FooBarRetainedRef` for when it is. But, that's super verbose. @dnoe for thoughts as well.